orca-core/src/main/java/com/netflix/spinnaker/orca/pipeline/util/RegionCollector.java

```diff
@@ -29,41 +28,45 @@
 
     List<Stage> descendentStages = stage.descendants();
 
-    Stream<Stage> descendentDeployStages = descendentStages.stream().filter(it -> "deploy".equals(it.getType()));
-    descendentDeployStages.forEach(deployStage -> {
-      List<Map> clusters = (List<Map>) deployStage.getContext().getOrDefault("clusters", new ArrayList<>());
+    descendentStages
+      .stream()
+      .filter(it -> "deploy".equals(it.getType()))
+      .forEach(deployStage -> {
+        List<Map> clusters = (List<Map>) deployStage.getContext().getOrDefault("clusters", new ArrayList<>());
 
-      for (Map cluster : clusters) {
-        if ((cluster.get("cloudProvider") != null) &&
-          cluster.get("cloudProvider").equals(stageCloudProviderType)) {
-          deployRegions.addAll(((Map) cluster.getOrDefault("availabilityZones", new HashMap())).keySet());
+        for (Map cluster : clusters) {
+          if ((cluster.get("cloudProvider") != null) &&
```

Comment:
Same null safe comparison as above could apply here ... 

```
if (stageCloudProviderType.equals(cluster.get("cloudProvider")) {
...
}
```

Not a big deal tho ... good use of `getOrDefault()`

Reply:
can also guarantee that `stageCloudProviderType` is always set, and then flip, makes is much cleaner i think:
`cloudProviderType.equals(cluster.get(...))`